tools/ci-scripts/windows/Install-Boost.ps1

```diff
@@ -0,0 +1,172 @@
+﻿<#
+.SYNOPSIS
+
+Downloads Boost via NuGet and returns the BOOST_ROOT and BOOST_LIBRARYDIR
+locations.
+
+.PARAMETER Version
+
+The Boost version to install.
+
+.PARAMETER Components
+
+The Boost binary components to install. The Boost headers are always installed.
+
+.PARAMETER VsNum
+
+Which MSVC compiler version to find libraries for, in Visual Studio version
+number format. (E.g., 12.0, 14.1).
+
+.PARAMETER OutputDirectory
+
+Where to install Boost. If not set, creates a directory in the TEMP folder.
+
+#>
+[CmdletBinding(SupportsShouldProcess=$True)]
+param
+(
+    [string]
+    $Version,
+
+    [ValidateSet('12.0', '14.0', '14.1')]
+    [string]
+    $VsNum,
+    
+    [string[]]
+    $Components = (
+        'boost_chrono',
+        'boost_date_time',
+        'boost_thread',
+        'boost_system',
+        'boost_unit_test_framework'),
+
+    [string]
+    $OutputDirectory = $null
+)
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+
+function Get-TempFolder
```

Comment:
Why not `System.IO.Path.GetTempFileName`, or simply append a GUID?

Reply:
✔️